src/main/java/com/mraof/minestuck/entity/underling/BasiliskEntity.java

```diff
@@ -1,151 +1,184 @@
-		this.goalSelector.addGoal(3, new CustomMeleeAttackGoal(this, 1.0F, false, 40, 1.2F));
-	}
-	
-	protected SoundEvent getAmbientSound()
-	{
-		return MSSoundEvents.ENTITY_BASILISK_AMBIENT;
-	}
-	
-	protected SoundEvent getHurtSound(DamageSource damageSourceIn)
-	{
-		return MSSoundEvents.ENTITY_BASILISK_HURT;
-	}
-	
-	protected SoundEvent getDeathSound()
-	{
-		return MSSoundEvents.ENTITY_BASILISK_DEATH;
-	}
-	
-	@Override
-	public GristSet getGristSpoils()
-	{
-		return GristHelper.generateUnderlingGristDrops(this, damageMap, 6);
-	}
-	
-	@Override
-	protected int getVitalityGel()
-	{
-		return random.nextInt(3) + 4;
-	}
-	
-	@Override
-	protected void onGristTypeUpdated(GristType type)
-	{
-		super.onGristTypeUpdated(type);
-		applyGristModifier(Attributes.MAX_HEALTH, 20 * type.getPower(), AttributeModifier.Operation.ADDITION);
-		applyGristModifier(Attributes.ATTACK_DAMAGE, 2.7 * type.getPower(), AttributeModifier.Operation.ADDITION);
-		this.xpReward = (int) (6 * type.getPower() + 4);
-	}
-	
-	@Override
-	public World getWorld()
-	{
-		return this.level;
-	}
-	
-	@Override
-	public void baseTick()
-	{
-		super.baseTick();
-		this.updatePartPositions();
-	}
-	
-	@Override
-	public boolean attackEntityFromPart(Entity entityPart, DamageSource source, float damage)
-	{
-		return this.hurt(source, damage);
-	}
-	
-	@Override
-	protected void doPush(Entity par1Entity)
-	{
-		if(par1Entity != this.tail)
-			super.doPush(par1Entity);
-	}
-	
-	@Override
-	public void absMoveTo(double par1, double par3, double par5, float par7, float par8)
-	{
-		super.absMoveTo(par1, par3, par5, par7, par8);
-		this.updatePartPositions();
-	}
-	
-	@Override
-	public void updatePartPositions()
-	{
-		if(tail == null)
-			return;
-		float f1 = this.yRotO + (this.yRot - this.yRotO);
-		double tailPosX = (this.getX() + Math.sin(f1 / 180.0 * Math.PI) * tail.getBbWidth());
-		double tailPosZ = (this.getZ() + -Math.cos(f1 / 180.0 * Math.PI) * tail.getBbWidth());
-		
-		tail.absMoveTo(tailPosX, this.getY(), tailPosZ, this.yRot, this.xRot);
-	}
-	
-	@Override
-	public void addPart(Entity entityPart, int id)
-	{
-		this.tail = (UnderlingPartEntity) entityPart;
-	}
-	
-	@Override
-	public void onPartDeath(Entity entityPart, int id)
-	{
-	
-	}
-	
-	@Override
-	public void die(DamageSource cause)
-	{
-		super.die(cause);
-		Entity entity = cause.getEntity();
-		if(this.dead && !this.level.isClientSide)
-		{
-			computePlayerProgress((int) (30 + 2.4 * getGristType().getPower())); //most basilisks stop giving xp at rung 32
-			firstKillBonus(entity, (byte) (Echeladder.UNDERLING_BONUS_OFFSET + 2));
-		}
-	}
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.PlayState;
+import software.bernie.geckolib3.core.builder.AnimationBuilder;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+
+public class BasiliskEntity extends UnderlingEntity implements IAnimatable {
```

Comment:
This class is no longer using the same code style as the rest of the project. Make sure that your code style settings is in line with the rest of the project (such as tab indents and brackets on a new line in most cases). I realize that this also applies to any new files introduced in this PR, so please reformat those as well.

Reply:
It could be a good idea to include the code styles in the version control, by default when i see tabs i always think of legacy code so i replace em as i go.

Reply:
can you share your code style file? I can add it as a project code style through this pr if you want

Reply:
Agreed, the code style should be included in the version control properly.